src/BigNumber.php

```diff
@@ -61,7 +61,7 @@ public static function of($value) : BigNumber
             return new BigInteger((string) $value);
         }
 
-        $value = (string) $value;
+        $value = self::convertToString($value);
```

Comment:
I'd rather only do this if `is_float()`. I prefer to keep assuming that strings use the default format with a dot as separator.

Reply:
BTW, types other than `float` should still be converted to string, so it should look like:

```php
if (is_float($value)) {
    // ...
} else {
    $value = (string) $value;
}
```